src/main/java/stsjorbsmod/cards/wanderer/Patron.java

```diff
@@ -36,13 +35,7 @@ public Patron() {
 
     @Override
     public void use(AbstractPlayer p, AbstractMonster m) {
-        addToBot(new DamageAction(m, new DamageInfo(p, damage)));
-
-        if (upgraded) {
-            addToBot(new DowngradeCardPermanentlyAction(this));
-        } else {
-            addToBot(new DestroyCardAction(this));
-        }
+        addToBot(new PatronAction(m, new DamageInfo(p, damage), this));
```

Comment:
I think I actually preferred the model where it was separate actions for damage vs the downgrade/destroy; I think it would make more sense for downgrade/destroy to own their own respective effect timings.

Reply:
It was necessary to do this because, like with Wrath, the downgrade/destroy should still occur if the card ended the combat. This was done following Watcher's upgrade card `LessonLearned`

Reply:
or were you thinking maybe something like PatronDowngradeAction and PatronDestroyAction both do damage, but do their respective downgrade/destroy?

Reply:
Looking over it again, the effect timings aren't actually tied to this action.